db/src/lib.rs

```diff
@@ -35,6 +32,34 @@ pub async fn create_user(pool: &PgPool, user_name: &str) -> color_eyre::Result<U
     Ok(result.user_id)
 }
 
+pub async fn add_url(pool: &PgPool, url: &str, allow_existing: &bool) -> color_eyre::Result<String> {
+    let exist_record = sqlx::query_as!(
+        ExistRecord,
+        "SELECT EXISTS(SELECT 1 FROM Page WHERE url = $1) as exists",
+        url
+    )
+    .fetch_one(pool)
+    .await?;
+
+    // exist_record might be true or false
+    if let Some(true) = exist_record.exists {
+        if *allow_existing {
```

Comment:
I used the dereferencing with the * operator to access the underlying value that a reference points to in Rust. This is necessary when you need to use the actual value held by the reference in expressions, conditions, or for passing to functions that require a value rather than a reference.